xarray/core/rolling_exp.py

```diff
@@ -56,7 +65,7 @@ def _get_center_of_mass(comass, span, halflife, alpha):
     return float(comass)
 
 
-class RollingExp:
+class RollingExp(Generic[T_DSorDA]):
```

Comment:
Great idea with `Generic`

Reply:
Yes that took quite a while to figure out but it doesn't work without it as the Type of `obj` does not bind to the function else. That should come in handy for some other classes.